src/components/editMeasure/details/cmsIdentifier/CmsIdentifier.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import {
+  InputLabel,
+  MadieTooltip,
+  Button,
+  ReadOnlyTextField,
+} from "@madie/madie-design-system/dist/react";
+
+export default function CmsIdentifier({ label, cmsId, model, onClick }) {
+  return (
+    <div
+      style={{
+        width: 1,
+        display: "flex",
+        flexDirection: "column",
+        flexGrow: 1,
+        fontFamily: "Rubik",
+        lineHeight: "19px",
+        fontSize: "16px",
+      }}
+    >
+      <div style={{ display: "flex", flexDirection: "row" }}>
+        <InputLabel for="cmsId">{label}</InputLabel>
+
+        <div>
+          <MadieTooltip
+            tooltipText="Once a CMS Identifier has been generated it may not be modified or removed for any draft or version of a measure."
+            id={`cms-id-tooltip`}
+          />
+        </div>
+      </div>
+
+      {cmsId > 0 ? (
+        <ReadOnlyTextField
+          tabIndex={0}
+          placeholder="CMS ID"
+          id="cmsId"
+          data-testid="cms-id-text-field"
+          inputProps={{ "data-testid": "cms-id-input" }}
+          size="small"
+          value={model.startsWith("QI-Core") ? `${cmsId}FHIR` : cmsId}
+          area-describedby="cms-id-tooltip"
+        />
+      ) : (
+        <Button
+          variant="secondary"
+          onClick={onClick}
+          data-testid="generate-cms-id-button"
+          area-describedby="cms-id-tooltip"
+          style={{
+            border: "none",
+          }}
+        >
+          <div style={{ display: "flex", alignItems: "center" }}>
+            <svg
```

Comment:
this could be made reusable. @mcmcphillips is already doing it. please take a look https://github.com/MeasureAuthoringTool/madie-patient/tree/develop/src/icons